### PR TITLE
Add AWS ECS provider

### PR DESCRIPTION
**Pull request by alekitto**
### What does this PR do?

Added ECS dynamic provider in v2 (#4674)

### More

- [x] Added/updated tests
- [x] Added/updated documentation

### Additional Notes

I used the v1 version of ECS provider as a base and adapted it based on v2 docker provider.

### CODE REVIEW

File: docs/content/providers/ecs.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+```
+
+```bash tab="CLI"
+--providers.ecs.region="us-east-1"
+--providers.ecs.accessKeyID="abc"
+--providers.ecs.secretAccessKey="123"
+# ...
+```
+
+If accessKeyID/secretAccessKey is not given credentials will be resolved in the following order:
```
Comment (SantoDE) [inline docs/content/providers/ecs.md:189]: given = provided maybe?

Comment (rtribotte) [reply docs/content/providers/ecs.md:189]: ok 👍

Comment (alekitto) [reply docs/content/providers/ecs.md]: Ok, I'll change it ASAP

**Comment by alekitto** [comment]
One small thing I noticed yesterday: the AutoDiscoverClusters setting is true by default. I think that should be false by default if the `clusters` list is not empty.

What do you think about it?

**Pull request merged**